src/main/java/uk/gov/hmcts/reform/em/stitching/service/callback/impl/CcdCallbackHandlerServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.em.stitching.service.callback.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.em.stitching.service.callback.CasePropertyFinder;
+import uk.gov.hmcts.reform.em.stitching.service.callback.CcdCallbackDto;
+import uk.gov.hmcts.reform.em.stitching.service.callback.CcdCallbackHandlerService;
+import uk.gov.hmcts.reform.em.stitching.service.callback.CcdCaseUpdater;
+
+@Service
+@Transactional
+public class CcdCallbackHandlerServiceImpl implements CcdCallbackHandlerService {
+
+    private CasePropertyFinder casePropertyFinder;
+
+    public CcdCallbackHandlerServiceImpl(CasePropertyFinder casePropertyFinder) {
+        this.casePropertyFinder = casePropertyFinder;
+    }
+
+    @Override
+    public JsonNode handleCddCallback(final CcdCallbackDto cmd, final CcdCaseUpdater ccdCaseUpdater) {
```

Comment:
not quite sure I understand why the variable name `cmd` is `cmd`. Also, why is the `ccdCaseUpdater` a method parameter rather than a class property?

Reply:
I tried to make the solution a generic one. The controller 'knows' the context of execution so that's why it sets the name of a property to retrieve and then the service that implements the change.